bundles/org.eclipse.jface/src/org/eclipse/jface/util/Throttler.java

```diff
@@ -54,34 +61,40 @@ public Throttler(Display display, Duration minWaitTime, Runnable runnable) {
 					"Minimum wait time in millis must be smaller than " + Integer.MAX_VALUE); //$NON-NLS-1$
 		}
 		int minWaitBetweenRunMillis = (int) minWaitTime.toMillis();
-		Runnable runner = () -> {
+		Runnable runner = () -> { // Always runs in Display Thread
 			scheduled.set(false);
 			runnable.run();
-			lastRunNanos = System.nanoTime();
+			long nanoTime = System.nanoTime();
+			if (nanoTime == 0) {
+				// prevent further 0 values.
+				nanoTime = -1;
```

Comment:
When can this happen? In such case, shouldn't we fall back to using 1000 * System,currentTimeMillis()?

Reply:
It may happen with chance of ~ 1/Long.MAXVALUE - which is most unlikely. When called in that unfortunate instant using the previous nanoTime  (-1) may execute the next event just 1 nanosecond too early - which feels OK.

Reply:
ok